datasets/dataset_esc50.py

```diff
@@ -0,0 +1,158 @@
+            y_test,
+            input_shape,
+            num_classes,
+            preprocess_dataset_labels,
+            generate_new_model_for_dataset,
+        )
+    return dataset_obj
+
+# Init dataset-specific functions
+def preprocess_dataset_labels(y):
+    num_classes = 50
+    y = to_categorical(y, num_classes)
+
+    return y
+
+
+def preprocess_dataset_inputs(x):
+    """
+    Compute the Mel-Frequency Cepstral Coefficients
+    :param x: iterator. Yield tuples, of audio and rate.
+    :return: Array of mfcc images.
+    """
+
+    features_list = []
+    for audio, rate in x:
+        mfccs = mfcc(y=audio, sr=rate, n_mfcc=40)
+        # mfccs_scaled = np.mean(mfccs.T, axis=0)
+        features_list.append(mfccs)
+    features = np.array(features_list)
+    return features.reshape(((features.shape[0],) + input_shape))
+
+
+# download train and test sets
+def load_data():
+    """
+    load the dataset. Note that the x are iterators which need to be preprocess
+    :return: (x_train, y_train), (x_test, y_test)
+    """
+    path = Path(__file__).resolve().parents[0]
+    repertoire = str(path) + '/local_data/esc50'
+    if not os.path.exists(repertoire):
+        os.makedirs(repertoire)
+        os.chdir(repertoire)
+        logger.info('ESC-50 dataset not found.')
+        _download_data(repertoire)
+    else:
+        logger.info('ESC-50 dataset found')
+        os.chdir(repertoire)
+    esc50_df = pd.read_csv('esc50.csv')
+    train, test = train_test_split(esc50_df, test_size=0.1, random_state=42)
+    y_train = train.target.to_numpy()
+    y_test = test.target.to_numpy()
+    x_train = (wav_load(repertoire + '/audio/' + file_name, sr=None) for file_name in train.filename.to_list())
+    x_test = (wav_load(repertoire + '/audio/' + file_name, sr=None) for file_name in test.filename.to_list())
+
+    os.chdir(str(Path(__file__).resolve().parents[1]))
+    return (x_train, y_train), (x_test, y_test)
+
+
+def _download_data(path):
+    """
+    Download the dataset, and unzip it. The dataset will be stored in the datasets/local_data directory, which is
+    gitignored
+
+    :param path: provided by load_data. Should be LOCAL_DIR/distributed-learning-contributivity/datasets/local_data/esc50
+    :return: None
+    """
+    logger.debug('Downloading it from https://github.com/karoldvl/ESC-50/')
+    urlretrieve('https://github.com/karoldvl/ESC-50/archive/master.zip', f'{path}/ESC-50.zip')
+    logger.debug('Extration at distributed-learning-contributivity/datasets/local_data/esc50')
+    with zipfile.ZipFile(f'{path}/ESC-50.zip') as package:
+        package.extractall(f'{path}/')
+
+    os.unlink(f'{path}/ESC-50.zip')
+    for src in glob.glob(f'{path}/ESC-50-master/audio'):
+        shutil.move(src, f'{path}/{os.path.basename(src)}')
+    shutil.move(f'{path}/ESC-50-master/meta/esc50.csv', f'{path}/esc50.csv')
+
+    shutil.rmtree(f'{path}/ESC-50-master')
+
+
+# Model structure and generation
+def generate_new_model_for_dataset():
```

Comment:
Do you have reference for this model (a blog post / a paper) or did you build it on your own ?

Reply:
Yes I have ! I add it